superset-frontend/src/explore/components/controls/FilterControl/AdhocFilterEditPopoverSimpleTabContent/index.tsx

```diff
@@ -0,0 +1,476 @@
+import AdhocFilter, {
+  EXPRESSION_TYPES,
+  CLAUSES,
+} from 'src/explore/components/controls/FilterControl/AdhocFilter';
+import { Input, SelectProps } from 'src/common/components';
+
+const SelectWithLabel = styled(Select)`
+  .ant-select-selector {
+    margin-bottom: ${({ theme }) => theme.gridUnit * 4}px;
+  }
+
+  .ant-select-selector::after {
+    content: '${(
+      pr: SelectProps<any> & {
+        labelText: string | boolean;
+      },
+    ) => pr.labelText || '\\A0'}';
+    display: inline-block;
+    white-space: nowrap;
+    color: ${({ theme }) => theme.colors.grayscale.light1};
+    width: max-content;
+  }
+`;
+
+export interface SimpleColumnType {
+  id: number;
+  column_name: string;
+  expression?: string;
+  type: string;
+  optionName?: string;
+  filterBy?: string;
+  value?: string;
+}
+
+export interface SimpleExpressionType {
+  expressionType: keyof typeof EXPRESSION_TYPES;
+  column: SimpleColumnType;
+  aggregate: keyof typeof AGGREGATES;
+  label: string;
+}
+export interface SQLExpressionType {
+  expressionType: keyof typeof EXPRESSION_TYPES;
+  sqlExpression: string;
+  label: string;
+}
+
+export interface MetricColumnType {
+  saved_metric_name: string;
+}
+
+export type ColumnType =
+  | SimpleColumnType
+  | SimpleExpressionType
+  | SQLExpressionType
+  | MetricColumnType;
+
+export interface Props {
+  adhocFilter: AdhocFilter;
+  onChange: (filter: AdhocFilter) => void;
+  options: ColumnType[];
+  datasource: {
+    id: string;
+    columns: SimpleColumnType[];
+    type: string;
+    filter_select: boolean;
+  };
+  partitionColumn: string;
+}
+export const useSimpleTabFilterProps = (props: Props) => {
+  const isOperatorRelevant = (operator: Operators, subject: string) => {
+    const column = props.datasource.columns?.find(
+      col => col.column_name === subject,
+    );
+    const isColumnBoolean =
+      !!column && (column.type === 'BOOL' || column.type === 'BOOLEAN');
+    const isColumnNumber = !!column && column.type === 'INT';
+    const isColumnFunction = !!column && !!column.expression;
+
+    if (operator && CUSTOM_OPERATORS.has(operator)) {
+      const { partitionColumn } = props;
+      return partitionColumn && subject && subject === partitionColumn;
+    }
+    if (operator === Operators.IS_TRUE || operator === Operators.IS_FALSE) {
+      return isColumnBoolean || isColumnNumber || isColumnFunction;
+    }
+    if (isColumnBoolean) {
+      return (
+        operator === Operators.IS_NULL || operator === Operators.IS_NOT_NULL
+      );
+    }
+    return !(
+      (props.datasource.type === 'druid' &&
+        TABLE_ONLY_OPERATORS.indexOf(operator) >= 0) ||
+      (props.datasource.type === 'table' &&
+        DRUID_ONLY_OPERATORS.indexOf(operator) >= 0) ||
+      (props.adhocFilter.clause === CLAUSES.HAVING &&
+        HAVING_OPERATORS.indexOf(operator) === -1)
+    );
+  };
+  const onSubjectChange = (id: string) => {
+    const option = props.options.find(
+      // @ts-ignore
```

Comment:
Why is the `ts-ignore` needed here? Can we type option as `ColumnType` to get around it?